internal/views/splits.go

```diff
@@ -479,9 +479,58 @@ func (n *Node) Unsplit() bool {
 	if n.parent.IsLeaf() {
 		return n.parent.Unsplit()
 	}
+
+	n.parent.flatten()
 	return true
 }
 
+// flattens the tree by removing unnecessary intermediate parents that have only one child
+// and handles the side effect of it
+func (n *Node) flatten() {
+	if n.parent == nil {
+		return
+	}
+
+	ind := 0
+	for i, c := range n.parent.children {
+		if c.id == n.id {
+			ind = i
+		}
+	}
+
+	if len(n.children) == 1 {
+		// Replace current node with child node to remove chained parent
+		n.parent.children[ind] = n.children[0]
+		n.parent.children[ind].parent = n.parent
+		if n.children[0].IsLeaf() {
+			n.parent.children[ind].Kind = n.Kind
+		} else {
+			// Flatten the child node that replaced the current node
+			n.parent.children[ind].flatten()
+		}
+	} else if !n.IsLeaf() && n.Kind == n.parent.Kind {
+		// If we have the same kind as the parent (cuased by previous flattening),
+		// replace this node with our children.
+		origsize := len(n.parent.children)
+
+		// Let's say we have 5 children and want to replace [2] with its children [a] [b] [c]
+		// [0] [1] [2] [3] [4] --> [0] [1] [a] [b] [c] [3] [4]
+		// insertcount will be `3 - 1 = 2` in this case
+		insertcount := len(n.children) - 1
+		n.parent.children = append(n.parent.children, make([]*Node, insertcount)...)
+		if ind != origsize-1 {
```

Comment:
if `ind == origsize-1`, the second slice has length 0, so `copy` is a no-op, so we can omit this if?